core/include/scipp/core/dataset.h

```diff
@@ -537,6 +537,7 @@ class SCIPP_CORE_EXPORT DatasetConstProxy {
   /// The returned proxy will not contain references to data items that do not
   /// depend on the sliced dimension.
   DatasetConstProxy slice(const Slice slice1) const {
+    expect::validSlice(*this, slice1.dim, slice1);
```

Comment:
Have you considered just using `Dataset::m_dims`? Does that eliminate the need for the unusual dimension merge code?

Reply:
Good idea, I'll try that.